### PR TITLE
Implement FTS table for PatientAddress

### PR DESCRIPTION
**Pull request by janhavisinghh**
https://app.shortcut.com/simpledotorg/story/8519/implement-search-query-for-overdue-patients

### CODE REVIEW

**Comment by janhavisinghh** [comment]
Current dependencies on/for this PR:
* master
  * **PR #3947** <a href="https://app.graphite.dev/github/pr/simpledotorg/simple-android/3947" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/simpledotorg/simple-android/3947?utm_source=stack-comment).

File: app/src/main/java/org/simple/clinic/patient/PatientAddress.kt [hunk 1]
```diff
@@ -103,3 +104,7 @@ data class PatientAddress(
     abstract fun getColonyOrVillages(): List<String>
   }
 }
+
+@Entity
+@Fts4(contentEntity = PatientAddress::class)
+data class PatientAddressFts(val uuid: UUID, val colonyOrVillage: String)
```
Comment (msasikanth) [inline app/src/main/java/org/simple/clinic/patient/PatientAddress.kt:110]: `PatientAddress` has `colonyOrVillage` as nullable. If existing records have a null property, the app will crash. Let's make this nullable as well?

File: app/src/main/java/org/simple/clinic/storage/migrations/Migration_103.kt [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+import androidx.sqlite.db.SupportSQLiteDatabase
+import org.simple.clinic.storage.inTransaction
+import javax.inject.Inject
+
+@Suppress("ClassName")
+class Migration_103 @Inject constructor() : Migration(102, 103) {
+
+  override fun migrate(database: SupportSQLiteDatabase) {
+    database.inTransaction {
+      execSQL("CREATE VIRTUAL TABLE IF NOT EXISTS `PatientAddressFts` USING FTS4(`uuid` TEXT NOT NULL, `colonyOrVillage` TEXT NOT NULL, content=`PatientAddress`)")
```
Comment (msasikanth) [inline app/src/main/java/org/simple/clinic/storage/migrations/Migration_103.kt:13]: Same comment as above for `colonyOrVillage` being nullable.

**Review by msasikanth — CHANGES_REQUESTED** [review]
See inline comments

File: app/src/main/java/org/simple/clinic/patient/PatientAddress.kt (hunk 1 above)
Comment (janhavisinghh) [reply app/src/main/java/org/simple/clinic/patient/PatientAddress.kt:110]: good catch

**Review by msasikanth — APPROVED** [review]

**Pull request merged**